operator-pipeline-images/operatorcert/entrypoints/github_add_comment.py

```diff
@@ -0,0 +1,167 @@
+It can take either a filename or a comment as input and can
+post the comment back to GitHub accordingly.
+"""
+import argparse
+from ast import arg
+import logging
+import json
+import os
+import http.client
+import sys
+import urllib.parse
+
+from operatorcert import github, store_results
+from operatorcert.logger import setup_logger
+
+LOGGER = logging.getLogger("operator-cert")
+
+
+def setup_argparser() -> argparse.ArgumentParser:  # pragma: no cover
+    """
+    Setup argument parser
+
+    Returns:
+        Any: Initialized argument parser
+    """
+    parser = argparse.ArgumentParser(
+        description="Github cli tool to add comment to PR or an issue."
+    )
+    parser.add_argument(
+        "--github-host-url",
+        default="api.github.com",
+        help="The GitHub host, default: api.github.com",
+    )
+    parser.add_argument(
+        "--request-url",
+        required=True,
+        help="The GitHub issue or pull request URL where we want to add a new comment.",
+    )
+    parser.add_argument(
+        "--comment-or-file",
+        required=True,
+        help="The actual comment to add or the filename containing comment to post.",
+    )
+    parser.add_argument(
+        "--comment-tag", help="An invisible tag to be added into the comment."
+    )
+    parser.add_argument(
+        "--replace",
+        default="false",
+        help="When a tag is specified, and `REPLACE` is `true`, look for a comment with a matching tag and replace it with the new comment.",
+    )
+    parser.add_argument("--verbose", action="store_true", help="Verbose output")
+    return parser
+
+
+def github_add_comment(
+    github_host_url: str,
+    request_url: str,
+    comment_or_file: str,
+    commen_tag: str,
+    replace: str,
+) -> None:
+    split_url = urllib.parse.urlparse(request_url).path.split("/")
+
+    # This will convert https://github.com/foo/bar/pull/202 to
+    # api url path /repos/foo/issues/
+    api_url = "{base}/repos/{package}/issues/{id}".format(
+        base="", package="/".join(split_url[1:3]), id=split_url[-1]
+    )
+
+    commentParamValue = comment_or_file
+    # check if parameter passed is a filename or not
+    if os.path.exists(commentParamValue):
+        commentParamValue = open(commentParamValue, "r").read()
+
+    # If a tag was specified, append it to the comment
+    if commen_tag != "":
+        commentParamValue += "<!-- {tag} -->".format(tag=commen_tag)
+    data = {
+        "body": commentParamValue,
+    }
+    # This is for our fake github server
+    if github_host_url.startswith("http://"):
+        conn = http.client.HTTPConnection(github_host_url.replace("http://", ""))
+    else:
+        conn = http.client.HTTPSConnection(github_host_url)
+
+    # If REPLACE is true, we need to search for comments first
+    matching_comment = ""
+    if replace == "true":
+        if not commen_tag:
+            print("REPLACE requested but no COMMENT_TAG specified")
```

Comment:
Instead of print() use `LOGGER.{info/debug/warning/error}`

Reply:
Also, the message makes sense in the old code. Please update a message to reflect actual script arguments.